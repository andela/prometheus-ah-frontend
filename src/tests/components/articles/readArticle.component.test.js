import configureMockStore from 'redux-mock-store';
import React from 'react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import DefaultReadArticle, { ReadArticle } from '../../../components/article/ReadArticle';

const mockStore = configureMockStore([thunk]);
describe('Read Article component', () => {
  let wrapper, store, props, initialState;
  beforeEach(() => {
    props = {
      article: {
        Tags: ['tag a', 'tag b'],
        User: {
          email: null,
          firstname: null,
          id: 5,
          image: null,
          lastname: null,
          reset_password_hash: null
        },
        body: '<p>The story started few years back</p>',
        createdAt: '2018-10-15T11:27:58.250Z',
        description: '<p>the description</p>',
        id: 71,
        readingTime: '1 min read',
        slug: 'new-article',
        title: 'new article',
        updatedAt: '2018-10-15T11:27:58.250Z',
        userId: 5

      },
      fetchSingleArticle: jest.fn(),
      match: {
        params: { slug: 'new-article' },
        url: '/articles/new-article'
      },
      location: { pathname: '/articles/new-article' },
      history: {}
    };
    initialState = {
      articleReducer: {
        articles: [
          {
            Tags: ['tag a', 'tag b'],
            User: {
              email: null,
              firstname: null,
              id: 5,
              image: null,
              lastname: null,
              reset_password_hash: null
            },
            body: '<p>The story started few years back</p>',
            createdAt: '2018-10-15T11:27:58.250Z',
            description: '<p>the description</p>',
            id: 71,
            readingTime: '1 min read',
            slug: 'new-article',
            title: 'Writing a new article',
            updatedAt: '2018-10-15T11:27:58.250Z',
            userId: 5
          }, {
            Tags: ['tag a', 'tag b'],
            User: {
              email: null,
              firstname: null,
              id: 5,
              image: null,
              lastname: null,
              reset_password_hash: null
            },
            body: '<p>The story started few years back when i was</p>',
            createdAt: '2018-10-15T11:27:58.250Z',
            description: '<p>the description of new article</p>',
            id: 71,
            readingTime: '1 min read',
            slug: 'new-article-edited',
            title: 'Writing a new article in a new dimension',
            updatedAt: '2018-10-15T11:27:58.250Z',
            userId: 5
          }
        ]
      }
    };
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <DefaultReadArticle {...props} />
      </Provider>
    );
  });

  it('should render Article page correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('ComponentDidMount', () => {
    const props = { // eslint-disable-line
      article: null,
      fetchSingleArticle: jest.fn(),
      match: {
        params: { slug: 'new-article' },
        url: '/articles/new-article'
      },
      location: { pathname: '/articles/new-article' },
      history: {}
    };
    wrapper = shallow(<ReadArticle store={store} {...props} />);
    expect(props.fetchSingleArticle).toHaveBeenCalled();
  });

  it('should display the necessary element', () => {
    expect(wrapper.find('div').exists()).toBe(true);
    expect(wrapper.find('div').length).toEqual(3);
  });

  it('ComponentDidMount', () => {
    const newProp = { ...props, artcile: null };
    wrapper = shallow(<ReadArticle {...newProp} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Loading')).toBeDefined();
  });
});
