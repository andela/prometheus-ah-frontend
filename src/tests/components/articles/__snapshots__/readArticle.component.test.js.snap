// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Read Article component ComponentDidMount 1`] = `
<div
  className="read-article my-5"
>
  <div
    className="row"
  >
    <div
      className="col-12 col-lg-8 offset-lg-2"
    >
      <div
        className="article-title"
      >
        <h1
          className="title"
        >
          new article
        </h1>
        <div
          className="d-flex flex-column align-items-start flex-sm-row"
        >
          <div
            className="mr-5 text-muted align-self-sm-center"
          >
            1 min read
          </div>
          <div
            className="text-muted mr-5 align-self-sm-center"
          >
            Oct 15, 2018
          </div>
          <div
            className="d-flex justify-content-around align-items-start"
          >
            <i
              className="mdi mdi-thumb-up-outline mr-3 align-top"
            >
              <span
                className="likes-num"
              >
                 30
              </span>
            </i>
            <a
              href="https://twitter.com/intent/tweet?url=http%3A%2F%2Flocalhost%2F&text=Check%20out%20this%20article%20from%20Authors%20Haven&hashtags=AuthorsHaven"
            >
              <i
                className="mdi mdi-twitter mr-3"
              />
            </a>
            <a
              href="https://www.facebook.com/dialog/share?app_id=344596382778592&display=page&href=http%3A%2F%2Flocalhost%2F&redirect_uri=http%3A%2F%2Flocalhost%2F"
            >
              <i
                className="mdi mdi-facebook mr-3"
              />
            </a>
            <a
              href="mailto:?Subject=Check%20out%20this%20article%20from%20Authors%20Haven Titled new article&body=http%3A%2F%2Flocalhost%2F"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                className="mdi mdi-email mr-3"
              />
            </a>
            <i
              className="mdi mdi-bookmark"
            />
          </div>
        </div>
        <p
          className="mt-3 description font-italic text-muted"
        >
          <p
            key="0"
          >
            the description
          </p>
        </p>
      </div>
    </div>
    <div
      className="col-12 mt-5 col-lg-8 offset-lg-2"
    >
      <p
        className="post-body"
      >
        <p
          key="0"
        >
          The story started few years back
        </p>
      </p>
    </div>
    <div
      className="col-12 mt-4 offset-lg-2"
    >
      <span
        className="btn tags"
      />
      <span
        className="btn tags"
      />
    </div>
    <div
      className="col-12 col-md-9 offset-md-1 mt-4 col-lg-6 offset-lg-3"
    >
      <div
        className="shadow-lg border-0 card p-3 p-lg-4"
      >
        <div
          className="d-flex mb-3"
        >
          <img
            alt=""
            className="img-fluid article-author-image rounded-circle"
            src="https://image.ibb.co/i48Wqf/paceholder.jpg"
          />
          <div
            className="ml-3 ml-md-3 ml-lg-4"
          >
            <p
              className="my-2 author-name"
            />
            <p
              className="author-bio"
            />
            <button
              className="btn follow-btn"
              type="button"
            >
              follow
            </button>
          </div>
        </div>
      </div>
      <div
        className="article-reactions mt-4 d-flex justify-content-around"
      >
        <i
          className="mdi mdi-thumb-up-outline"
        >
          <span
            className="likes-num"
          >
             30
          </span>
        </i>
        <a
          href="https://twitter.com/intent/tweet?url=http%3A%2F%2Flocalhost%2F&text=Check%20out%20this%20article%20from%20Authors%20Haven&hashtags=AuthorsHaven"
        >
          <i
            className="mdi mdi-twitter"
          />
        </a>
        <a
          href="https://www.facebook.com/dialog/share?app_id=344596382778592&display=page&href=http%3A%2F%2Flocalhost%2F&redirect_uri=http%3A%2F%2Flocalhost%2F"
        >
          <i
            className="mdi mdi-facebook"
          />
        </a>
        <a
          href="mailto:?Subject=Check%20out%20this%20article%20from%20Authors%20Haven Titled new article&body=http%3A%2F%2Flocalhost%2F"
          rel="noopener noreferrer"
          target="_blank"
        >
          <i
            className="mdi mdi-email"
          />
        </a>
        <i
          className="mdi mdi-bookmark"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Read Article component should render Article page correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ReadArticle)
    article={
      Object {
        "Tags": Array [
          "tag a",
          "tag b",
        ],
        "User": Object {
          "email": null,
          "firstname": null,
          "id": 5,
          "image": null,
          "lastname": null,
          "reset_password_hash": null,
        },
        "body": "<p>The story started few years back</p>",
        "createdAt": "2018-10-15T11:27:58.250Z",
        "description": "<p>the description</p>",
        "id": 71,
        "readingTime": "1 min read",
        "slug": "new-article",
        "title": "new article",
        "updatedAt": "2018-10-15T11:27:58.250Z",
        "userId": 5,
      }
    }
    fetchSingleArticle={[MockFunction]}
    history={Object {}}
    location={
      Object {
        "pathname": "/articles/new-article",
      }
    }
    match={
      Object {
        "params": Object {
          "slug": "new-article",
        },
        "url": "/articles/new-article",
      }
    }
  >
    <ReadArticle
      fetchSingleArticle={[Function]}
      history={Object {}}
      location={
        Object {
          "pathname": "/articles/new-article",
        }
      }
      match={
        Object {
          "params": Object {
            "slug": "new-article",
          },
          "url": "/articles/new-article",
        }
      }
    >
      <div
        className="read-article my-5"
      >
        <div
          className="row"
        >
          <Loading>
            <div
              className="loading-center"
            >
              <img
                alt="Loading..."
                className="text"
                src={Object {}}
              />
            </div>
          </Loading>
        </div>
      </div>
    </ReadArticle>
  </Connect(ReadArticle)>
</Provider>
`;
